src/App.tsx

```diff
@@ -165,10 +167,22 @@ const App = () => {
     };
   }, [onSetAutomation]);
 
-  const onSetMainProcessEnv = useCallback((_: any, args: any) => {
-    setMainProcessEnv(args.mainProcessEnv);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  const onSetMainProcessEnv = useCallback(
+    (_: any, args: any) => {
+      if (args && typeof args.serializedMainProcessEnv === 'string') {
+        setMainProcessEnv(JSON.parse(args.serializedMainProcessEnv));
+      } else {
+        dispatch(
```

Comment:
could this be in a try/catch statement around the JSON.parse?

Reply:
sure, will do